### DISCUSSION TITLE
Error: Could not send content to Obsidian: (Error Code 40400) Not Found

### DISCUSSION THREAD

**Discussion by j-peeters** [coddingtonbear/obsidian-web#56] [Q&A] [answered]
I get the following error message when I try to save a note to Obsidian. 

<img width="617" alt="20221024 Arc Obsidian Web error@2x" src="https://user-images.githubusercontent.com/4502360/197502934-fa5f7a59-2b57-4b3b-ba4e-0e14bcb6abad.png">

Obsidian is properly installed and I have a Vault 'Notes' in a folder named 'Obsidian' in my user folder 
/Users/jpeeters/Obsidian/Notes/

Does anyone know what this error code means and possibly how to solve it? Thanks.

**Comment by coddingtonbear** [comment]
This error code means that the path you've configured in your template is pointing at a route that does not exist.  I'm afraid I can't offer any further suggestions without more information about the template you've set up.

**Reply by huyz** [reply]
This happened to me in the course of trying to figure out how to use this plugin and I was very confused. Naturally, I tried "Append to existing note". I assumed it would append to the _active_ (focused) note that I had up and open in Obsidian. Now that I understand how this plugin works, I can see that was an incorrect assumption.

Here are some suggestions:
- disable the ability of the user to select the "Append to existing note" menu item since there's no URL
- change "existing" to "matching"
- give a more descriptive error message: "The URL specified in the template was not found", and in the case of a blank URL "URL needs to be specified in the template".
- Have Local REST API expose a route that supports "Append to focused note"

**Reply by j-peeters** [reply]
Thanks for your replies @coddingtonbear and @huyz.
I also had the idea that it might have to do with my URL setup.
I have tried many configurations but didn't find the right one.

My folder in which the plugin should store the note is
`/Users/jpeeters/Obsidian/Notes/Sources`

In this case which URL should be entered in the template?

My Obsidian Local REST API is setup like this:

<img width="1129" alt="20221025 Obsidian Web Options 2@2x" src="https://user-images.githubusercontent.com/4502360/197756179-674679ab-ee0a-4ddc-a983-39383cfaeee4.png">

And I currently have it like this when I choose 'New Note'

<img width="1040" alt="20221025 Obsidian Web options 2 2@2x" src="https://user-images.githubusercontent.com/4502360/197756513-fd8ab721-f628-401a-89b8-4d05f4a5db53.png">

Thanks for your continued help and patience ;-)

**Reply by coddingtonbear** [reply]
I think if you look at the original version of that template, you'll find a slightly different value in the `API URL` field:

![image](https://user-images.githubusercontent.com/527661/197790380-824a1f8e-e7c8-43a1-b6be-d417dbe2fce4.png)

What I'd recommend doing is restoring that field's value to what you see in the above screenshot (`/vault/{{filename page.title}}.md`) and see where your file is written within you vault (hint: it'll be at the root folder of it).

The "API URL" you enter into that field is one of the API URLs provided by obsidian-local-rest-api (see https://coddingtonbear.github.io/obsidian-local-rest-api/ for the documented routes), it's not just a path on your filesystem.

**Comment by j-peeters** [comment, marked as answer]
Thanks, that did it.

I guess I started trying adding URL info when I got the (what I supposed was an error) shield (which was just the insecure HTTP warning). Reverting it back to `vault` saved notes again.

It might be an idea to write something under the API URL field that mentions that this is not a path like in the OS. I got the idea from it that `vault` was a placeholder for the real name of your vault and that it has to be replaced.

But I'm glad everything works now, it's the most fluent Obsidian Clipper I've come across. Thanks!
